### PR TITLE
Unify shell.nix usage

### PR DESCRIPTION
**Pull request by cole-h**
This allows us to drop reliance on environment variables such as
NIXPKGS_ALLOW_UNFREE and NIX_PATH.

---

Untested (for obvious reasons).

### CODE REVIEW

**Comment by cole-h** [comment]
cc @grahamc

**Comment by grahamc** [comment]
I wonder if we should relocate the new shell.nix and nixpkgs pinning to the root of the repo, and delete the old bits?

**Comment by cole-h** [comment]
I was hoping you'd say that. Done in 2c441ce772a7dd772af4aa051785d93562d2fbbe, PTAL and make sure I wasn't over- (or under-)eager in my deletions.

File: build/clone.sh [hunk 1]
```diff
@@ -1,9 +1,7 @@
 #!/usr/bin/env nix-shell
-#!nix-shell -i bash ../shell.nix
+#!nix-shell -i bash ../terraform/shell.nix
```
Comment (grahamc) [inline build/clone.sh:2]: I think this needs a minor fixup?

```suggestion
#!nix-shell -i bash ../shell.nix
```

Comment (cole-h) [reply build/clone.sh]: There were a few other instances (e.g. `ci/terraform-apply.sh`) that I fixed up as well and amended.

File: shell.nix [hunk 2]
```diff
@@ -1,26 +1,91 @@
...
+        # if the versions are not provided via file paths.
+        postBuild = ''
+          mv $NIX_BUILD_TOP/go/bin/${pname}{,_v${version}}
+        '';
+      })
     ]))
- ];
-
-  HISTFILE = "${toString ./.}/.bash_hist";
-  NIX_PATH = "nixpkgs=${pkgs.path}:ofborg-infra=${toString ./.}";
```
Comment (cole-h) [inline shell.nix:25]: Do we want to resurrect this into the new shell.nix?

Comment (grahamc) [reply shell.nix:25]: Hmm... Yeah, let's...

Comment (cole-h) [reply shell.nix:25]: Done.

**Pull request merged**